cni.go

```diff
@@ -129,6 +100,8 @@ func (c *libcni) Setup(id string, path string, opts ...NamespaceOpts) (*CNIResul
 		return nil, err
 	}
 	var results []*current.Result
+	c.RLock()
+	defer c.RUnlock()
 	for _, network := range c.networks {
 		r, err := network.Attach(ns)
```

Comment:
Is this really a read action? What happens on concurrent attach?

Reply:
the network object itself is not changing. So technically a read